File: MergeTwoSortedLists/MergeTwoSortedLists.java
/*
public class ListNode {
   public int val; 
   public ListNode next;
   public ListNode(int x) { val = x; next = null; }
}
*/
import crio.ds.List.ListNode;
import java.io.*;
import java.util.*;

class MergeTwoSortedLists {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        ArrayList<Integer> list1 = new ArrayList<Integer>();
        for(int i = 0 ; i < n ; ++i) {
            int x = scanner.nextInt();
            list1.add(x);
        }
        int m = scanner.nextInt();
        ArrayList<Integer> list2 = new ArrayList<Integer>();
        for(int i = 0 ; i < m ; ++i) {
            int x = scanner.nextInt();
            list2.add(x);
        }
        scanner.close();

        ListNode list1Head = ListNode.createList(list1);
        ListNode list2Head = ListNode.createList(list2);
        ListNode resultHead = new Solution().mergeTwoLists(list1Head, list2Head);

        List<Integer> res = ListNode.extractList(resultHead);
        for(int element: res) {
            System.out.print(element + " ");
        }
    }
}

File: MergeTwoSortedLists/MergeTwoSortedLists.cpp
/*
class ListNode {
    public:
	int val;
	ListNode* next;

	ListNode() {}

	ListNode(int x) {                      
	    val = x;                       
	    next = NULL;
	}
};
*/
#include <bits/stdc++.h>
using namespace std;
#include "Solution.cpp"

int main() {
	int m, n;
	cin >> m;
	vector<int> nums1(m);

	for(int i = 0 ; i < m ; ++i) {
            cin >> nums1[i];
        }
    	cin >> n;
	vector<int> nums2(n);
	for(int i = 0 ; i < n ; ++i) {
            cin >> nums2[i];
        }

	ListNode* l1 = ListNode().createList(nums1);
    	ListNode* l2 = ListNode().createList(nums2);

    	ListNode* resultHead = mergeTwoLists(l1, l2);

	vector<int> resultList = ListNode().extractList(resultHead);
        for(int i = 0 ; i < resultList.size() ; ++i) {
            cout << resultList[i] << " ";
        }
	return 0;
}
